src/Voron/Data/Tables/Table.cs

```diff
@@ -331,6 +331,12 @@ public bool IsOwned(long id)
 
         public void Delete(long id)
         {
+            if (IsOwned(id) == false)
+            {
+                Debug.Assert(false, $"Trying to delete a values (id:{id}) from the wrong table ('{Name}')");
```

Comment:
what that debug assert gives you if you are throwing here anyway?

Reply:
I want immediately to kill the test here, in order to expose the error and not to have at the end of the test something like: `Expected:0, Actual:1`